src/seraphis_main/contextual_enote_record_types.cpp

```diff
@@ -133,13 +399,83 @@ const SpEnoteSpentContextV1& spent_context_ref(const ContextualRecordVariant &va
         using variant_static_visitor::operator();  //for blank overload
         const SpEnoteSpentContextV1& operator()(const LegacyContextualEnoteRecordV1 &record) const
         { return record.spent_context; }
+        const SpEnoteSpentContextV1& operator()(const LegacyContextualEnoteRecordV2 &record) const
+        { return record.spent_context; }
         const SpEnoteSpentContextV1& operator()(const SpContextualEnoteRecordV1 &record) const
         { return record.spent_context; }
     };
 
     return variant.visit(visitor());
 }
 //-------------------------------------------------------------------------------------------------------------------
+bool is_older_than(const LegacyEnoteOriginContextV1 &context, const LegacyEnoteOriginContextV1 &other_context)
+{
+    // 1. origin status (higher statuses are assumed to be 'older')
+    if (context.origin_status > other_context.origin_status)
+        return true;
+    if (context.origin_status < other_context.origin_status)
+        return false;
+
+    // 2. block index
+    if (context.block_index < other_context.block_index)
+        return true;
+    if (context.block_index > other_context.block_index)
+        return false;
+
+    // note: don't assess the tx output index
+
+    // 3. enote ledger index
+    if (context.enote_ledger_index < other_context.enote_ledger_index)
+        return true;
+    if (context.enote_ledger_index > other_context.enote_ledger_index)
+        return false;
+
+    // 4. block timestamp
+    if (context.block_timestamp < other_context.block_timestamp)
+        return true;
+    if (context.block_timestamp > other_context.block_timestamp)
+        return false;
+
+    return false;
+}
+//-------------------------------------------------------------------------------------------------------------------
+bool is_older_than(const LegacyEnoteOriginContextV2 &context, const LegacyEnoteOriginContextV2 &other_context)
+{
+    // 1. origin status (higher statuses are assumed to be 'older')
+    if (context.origin_status > other_context.origin_status)
+        return true;
+    if (context.origin_status < other_context.origin_status)
+        return false;
+
+    // 2. block index
+    if (context.block_index < other_context.block_index)
+        return true;
+    if (context.block_index > other_context.block_index)
+        return false;
+
+    // note: don't assess the tx output index
+
+    // 3. enote zero amount ledger index
+    if (context.rct_enote_ledger_index < other_context.rct_enote_ledger_index)
```

Comment:
From this conversation https://github.com/UkoeHB/monero/pull/40#discussion_r1536557163
But it does make sense to compare `rct_enote_ledger_index` here, because even if we have a enote with duplicate onetime address, both use the zero amount index, or am I missing something?